### ISSUE TITLE
« 'utf8' codec can't decode byte 0xe8 in position 13: invalid continuation byte » errors

### ISSUE THREAD

**Issue by BleachThemAll** [bleachbit/bleachbit#328]
I'm using Bleachbit 2.0 (GUI) on Windows 7 64bits.

This bug was first reported by me here :
https://www.bleachbit.org/comment/3006#comment-3006

This problem occurs only when I launch Bleachbit using "BleachBit No UAC" in the Windows start menu, never when I use "BleachBit" in the start menu.

During deletion (not during preview), I get those red error messages (the beginning in french language means "Error while running [...]") :

Erreur lors de l’exécution de « winapp2_multimedia » : « 'utf8' codec can't decode byte 0xe8 in position 13: invalid continuation byte »

Erreur lors de l’exécution de « system » : « 'utf8' codec can't decode byte 0xe8 in position 13: invalid continuation byte »

Erreur lors de l’exécution de « winapp2_utilities » : « 'utf8' codec can't decode byte 0xe8 in position 13: invalid continuation byte »

Erreur lors de l’exécution de « winapp2_internet » : « 'utf8' codec can't decode byte 0xe8 in position 13: invalid continuation byte »

Erreur lors de l’exécution de « internet_explorer » : « 'utf8' codec can't decode byte 0xe8 in position 13: invalid continuation byte »

Erreur lors de l’exécution de « winapp2_mozilla » : « 'utf8' codec can't decode byte 0xe8 in position 13: invalid continuation byte »

I do not get any error message in bleachbit.exe.log .
The boxes causing errors are :

Erreur lors de l’exécution de « winapp2_multimedia » : in the Multimedia section,
Adobe Flash Player

Erreur lors de l’exécution de « system » : in the System section,
Fichiers temporaires (Temporary files)
Journaux (Logs ? Journals ? in english)

Erreur lors de l’exécution de « winapp2_utilities » : in the winapp2 Utilities section,
Dell Logs
Intel iCLS Client
Intel Installation Logs

Erreur lors de l’exécution de « winapp2_internet » : in the winapp2 internet section,
Java

Erreur lors de l’exécution de « internet_explorer » : in the internet explorer section,
Fichiers temporaires (Temporary Files)

Erreur lors de l’exécution de « winapp2_mozilla » : in the winapp2 mozilla section,
Firefox Logs
Stored HSTS Cookies Extras

Sometimes some of the errors do not appear any more, such as currently the winapp2_multimedia one caused by Flash and the winapp2_internet one caused by Java (maybe because there's nothing to delete there currently ?).

I also tried and am currently using BleachBit version 2.0.689 after being suggested so and saw no change.

**Comment by szc126** [comment]
I have the same problem, using 2.0, on Windows 7 64bit in a Japanese locale (could it be an encoding problem?).

Longer error message, an excerpt from bleachbit_console.exe:
```
Traceback (most recent call last):
  File "logging\__init__.pyo", line 861, in emit
  File "logging\__init__.pyo", line 734, in format
  File "logging\__init__.pyo", line 465, in format
  File "logging\__init__.pyo", line 329, in getMessage
UnicodeDecodeError: 'utf8' codec can't decode byte 0x83 in position 10: invalid
start byte
Logged from file Worker.pyo, line 97
Exception while running operation 'windows_explorer': ''utf8' codec can't decode
 byte 0x83 in position 10: invalid start byte'
Traceback (most recent call last):
  File "bleachbit\Worker.pyo", line 338, in run_operations
  File "bleachbit\Worker.pyo", line 150, in clean_operation
  File "bleachbit\Worker.pyo", line 97, in execute
  File "logging\__init__.pyo", line 1193, in error
  File "logging\__init__.pyo", line 1286, in _log
  File "logging\__init__.pyo", line 1296, in handle
  File "logging\__init__.pyo", line 1336, in callHandlers
  File "logging\__init__.pyo", line 759, in handle
  File "bleachbit\GUI.pyo", line 238, in emit
  File "logging\__init__.pyo", line 329, in getMessage
UnicodeDecodeError: 'utf8' codec can't decode byte 0x83 in position 10: invalid
start byte
```

The part that changes is ``while running operation '<whatever>'``, such as `windows explorer`, `winapp2_windows`, `internet_explorer`.

**Comment by az0** [comment]
@szc126 The traceback is helpful. It looks like there is an encoding error related to the display of an error message, probably _permission denied_ or _file not found_. I will take a closer look

**Comment by az0** [comment]
Thank you both for the bug report. The bug was related to improper handling of non-ASCII when trying to display an error message that was either "access denied" or "does not exist."

In about ten minutes, you can download a binary with a fix from the [BleachBit CI](https://ci.bleachbit.org/) site. You need version 2.0.724 or newer.

**Comment by az0** [comment]
The fix was published in [BleachBit 2.1 beta](https://www.bleachbit.org/news/bleachbit-21-beta). Please try it out, and let me know how it goes. 

Soon the fix will be published also in BleachBit 2.2 final.

**Comment by fjrg76mx** [comment]
Got it! I'll try it ASAP. Thanks for your effort ;)

**Comment by fjrg76mx** [comment]
Tested and it DOES NOT fix the problem:

```
$ bleachbit --version

BleachBit version 2.1
Copyright (C) 2008-2018 Andrew Ziem.  All rights reserved.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

$ bleachbit
Traceback (most recent call last):
  File "/usr/bin/bleachbit", line 38, in <module>
    import gtk
  File "/usr/lib/python2.7/dist-packages/gtk-2.0/gtk/__init__.py", line 40, in <module>
    from gtk import _gtk
ImportError: /usr/lib/x86_64-linux-gnu/libharfbuzz.so.0: undefined symbol: FT_Get_Var_Blend_Coordinates
$
``` 
I downloaded the "bleachbit_2.1_all_ubuntu1810.deb" package for my Linux Mint 19.1 box.

Somewhere I read the problem lies in the Infinality font rendering software. I'm using it in all my computers but one in which I tried it on a fresh installation, and it worked neatly.

**Comment by az0** [comment]
You could test this fix in the newly-released [BleachBit 2.3 beta](https://www.bleachbit.org/news/bleachbit-23-beta)
